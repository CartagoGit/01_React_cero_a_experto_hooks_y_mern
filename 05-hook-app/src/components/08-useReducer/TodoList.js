import React from "react";
import { TodoListitem } from "./TodoListitem";

export const TodoList = ({ todos, handleDelete, handleToggle }) => {
	return (
		<ul className='list-group list-group-flush'>
			{todos.map((todo, i) => (
				<TodoListitem
					key={todo.id}
					todo={todo}
					handleDelete={handleDelete}
					handleToggle={handleToggle}
					index={i}
				/>
			))}
		</ul>
	);
};
