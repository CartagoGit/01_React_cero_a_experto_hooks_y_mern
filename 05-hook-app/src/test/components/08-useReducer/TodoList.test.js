import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { TodoList } from "../../../components/08-useReducer/TodoList";
import { demoTodos } from "../../fixtures/demoTodos";

describe("Pruebas en <TodoList />>", () => {
	const handleDelete = jest.fn();
	const handleToggle = jest.fn();
	let wrapper;

	const fillWrapper = () => {
		wrapper = shallow(
			<TodoList
				todos={demoTodos}
				handleDelete={handleDelete}
				handleToggle={handleToggle}
			/>
		);
	};
	beforeEach(() => fillWrapper());

	test("Deberia mosrarse correctamente", () => {
		expect(wrapper).toMatchSnapshot();
	});

    test('Deberia tener el numero de list items ', () => {
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));
        expect(wrapper.find('TodoListItem').at(0).prop('handleToggle')).toEqual(expect.any(Function));
    });
    
});
