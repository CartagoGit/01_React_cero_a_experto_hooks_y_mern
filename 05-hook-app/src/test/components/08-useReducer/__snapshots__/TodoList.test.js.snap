// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList />> Deberia mosrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="123"
    todo={
      Object {
        "desc": "Aprender react",
        "done": false,
        "id": 123,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="234"
    todo={
      Object {
        "desc": "Tocarme las narices",
        "done": false,
        "id": 234,
      }
    }
  />
</ul>
`;
